### PR TITLE
feat: zip tasks load and display

### PR DESCRIPTION
**Pull request by TheRealSeber**

### CODE REVIEW

File: src/lib/components/taskFile.svelte.ts [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+	let inFiles = inFolder.filter((_, file) => isValidInFile(inFolderPath, file.name));
+	let outFiles = outFolder.filter((_, file) => isValidOutFile(outFolderPath, file.name));
+
+	if (inFiles.length !== outFiles.length) {
+		throw new Error('The number of input and output files is not equal');
+	}
+
+	for (let i = 0; i < inFiles.length; i++) {
+		const inFileName = inFiles[i].name;
+		const outFileName = outFiles.find((file) => file.name === inFileName.replaceAll('in', 'out'));
```
Comment (HermanPlay) [inline src/lib/components/taskFile.svelte.ts]: replaceAll strategy doesnt work if path contains "in". Forexample: _inversions/in/in4.txt_. After replace we have: _outversions/out/out4.txt_

**Review by HermanPlay — CHANGES_REQUESTED** [review]

File: src/lib/components/taskFile.svelte.ts (hunk 1 above)
Comment (TheRealSeber) [reply src/lib/components/taskFile.svelte.ts]: bruh

File: src/lib/server/db/schema.ts [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+	id: serial('id').primaryKey(),
+	taskId: integer('task_id').references(() => tasks.id),
+	order: integer('order').notNull(),
+	timeLimit: real('time_limit').notNull(),
+	memoryLimit: real('memory_limit').notNull()
+});
+
+export const submissionResults = pgTable('submission_results', {
+	id: serial('id').primaryKey(),
+	userSolutionId: integer('user_solution_id').references(() => submissions.id),
```
Comment (HermanPlay) [inline src/lib/server/db/schema.ts]: userSolutionId -> submissionId

File: src/lib/server/db/schema.ts [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+	statusCode: userSolutionResultStatusEnum('status_code').notNull(),
+	message: varchar('message', { length: 255 }),
+	createdAt: timestamp('created_at', { withTimezone: true }).defaultNow()
+});
+
+export const testResults = pgTable('test_results', {
+	id: serial('id').primaryKey(),
+	userSolutionResultId: integer('user_solution_result_id').references(() => submissionResults.id),
```
Comment (HermanPlay) [inline src/lib/server/db/schema.ts]: userSolutionResultId -> submissionResultId

File: src/lib/server/db/schema.ts [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+		userId: integer('user_id').references(() => users.id)
+	},
+	(table) => ({
+		pk: primaryKey({ columns: [table.taskId, table.userId] })
+	})
+);
+
+export const queueMessages = pgTable('queue_messages', {
+	id: text('id').primaryKey(),
+	userSolutionId: integer('user_solution_id').references(() => submissions.id),
```
Comment (HermanPlay) [inline src/lib/server/db/schema.ts]: userSolutionId -> submissionId

File: src/lib/server/db/schema.ts [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+	integer,
+	timestamp,
+	pgEnum,
+	real,
+	boolean,
+	varchar,
+	primaryKey
+} from 'drizzle-orm/pg-core';
+
+export const userSolutionStatusEnum = pgEnum('user_solution_status', [
```
Comment (HermanPlay) [inline src/lib/server/db/schema.ts]: userSolutionStatusEnum -> submissionStatusEnum

File: src/lib/server/db/schema.ts [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+	'submitted',
+	'queued',
+	'executed'
+]);
+export const languageTypeEnum = pgEnum('language_type', ['CPP', 'C', 'Python']);
+export const userSolutionResultStatusEnum = pgEnum('user_solution_result_status', [
```
Comment (HermanPlay) [inline src/lib/server/db/schema.ts]: userSolutionResultStatusEnum -> submissionResultStatusEnum

**Review by HermanPlay — APPROVED** [review]

**Pull request merged**